### DISCUSSION TITLE
Prioritize implementation of back-compat flags

### DISCUSSION THREAD

**Discussion by shueybubbles** [microsoft/go-sqlcmd#292] [General]
Here's the list of missing flags for "legacy" mode of sqlcmd as of 2023-03-08:

```
-D (DSN name)
-e (echo input)
-f codepage | i:codepage[,o:codepage] | o:codepage[,i:codepage]
-g (enable column encryption)
-j (Print raw error messages)
-k[1 | 2] (remove or replace control characters)
-p[1] (print statistics, optional colon format)
-t query_timeout
-z new_password
-Z new_password (and exit)
```

These flags are partially implemented:
```
-r[0 | 1] (msgs to stderr)
-X[1] (disable commands, startup script, environment variables, optional exit)
```

Please vote on the flags you need implemented to have go-sqlcmd successfully replace the ODBC sqlcmd in your environment.

Some caveats:
- Column encryption isn't supported by the `go-mssqldb` driver so we can't implement `-g` yet. Support is planned for 2023.
- There's no concept of `DSN` outside of ODBC so `-D` can't be treated the same way as before. Would it be reasonable to have `-D` be our way to connect the legacy mode to the config file which is managed by `sqlcmd config`? Perhaps `-D` could accept the name of a connection in the config file and get the connection parameters from it.

**Comment by dzsquared** [comment]
A vote for -y to set display width beyond 256 characters for variable content.  My specific use case was access to XML returned from SQL

**Reply by shueybubbles** [reply]
you can set `SQLCMDMAXVARTYPEWIDTH` environment variable as the workaround.

**Reply by shueybubbles** [reply]
> A vote for -y to set display width beyond 256 characters for variable content. My specific use case was access to XML returned from SQL

We are also missing the `xml` command to put sqlcmd in XML output mode.

**Comment by shueybubbles** [comment]
#200 votes for `-A`
